### PR TITLE
Fixed interaction between dynamic texture and context.clip()

### PR DESCRIPTION
**Pull request by SergioRZMasson**
## Overview 
Allows Dynamic Texture context ```context.drawImage``` ```context.clearRect``` and ```context.fillRect``` to properly work with ```context.clip```.

## Implementation

When calling ```Babylon::Polyfills::Internal::Context::DrawImage``` the ```nvgBeginPath``` is been called every time. This will discard any path that was previously set by the user. The correct behavior should be to not call  ```nvgBeginPath``` if ```Babylon::Polyfills::Internal::Context::Clip``` was previously called. 

Also, in ```Babylon::Polyfills::Internal::Context::Clip``` the ```m_rectangleClipping``` is been used to set the Scissor values even when  ```m_rectangleClipping``` was never set before (with default values 0, 0, 0, 0). This will make some of the following operations to produce no result. The correct behavior should be to only use ```m_rectangleClipping``` if the were set, otherwise use the width and height of the render target. 

Finally, when creating Babylon::Polyfills::Internal::NativeCanvasImage the resulting data might be in RGB instead of RGBA. However, during ```NativeCanvasImage::CreateNVGImageForContext``` NVG always assumes the format to be RGBA, which might cause a memory out of bounds read if the image was originally RGB. This PR forces ```bimg::imageParse``` to always convert the image to RGBA. 

## More information
Allows the following code to work in Babylon Native: https://playground.babylonjs.com/#FU0ES5#27
Fixes issue #1184

### CODE REVIEW

File: Apps/ValidationTests/Scripts/config.json [hunk 1]
```diff
@@ -1,383 +1,389 @@
...
-            "replace": "__folder__, Texture_Sampler, __page__, 1, __disableSRGBBuffers__, 0",
-            "referenceImage": "gltfTextureSampler1.png"
-        },
-        {
-            "title": "GLTF Alien",
-            "playgroundId": "#XN37SR#5",
-            "referenceImage": "gltfAlien.png"
-        }
-    ]
+  "root": "https://cdn.babylonjs.com",
```
Comment (CedricGuillemet) [inline Apps/ValidationTests/Scripts/config.json:2]: Can you change the tab so it's easier to see the change?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -92,8 +92,8 @@ jobs:
     parameters:
       name: Ubuntu_Clang9_V8
```
Comment (CedricGuillemet) [inline azure-pipelines.yml:93]: Don't forget to change the job name

File: Apps/ValidationTests/Scripts/config.json [hunk 3]
```diff
@@ -1,383 +1,389 @@
...
+  "tests": [
+    {
+      "title": "Light Projection Texture",
```
Comment (CedricGuillemet) [inline Apps/ValidationTests/Scripts/config.json:5]: "Light Projection Texture" It's in the config.json 2 times. is it expected?

File: Apps/ValidationTests/Scripts/config.json [hunk 4]
```diff
@@ -1,383 +1,389 @@
...
-            "referenceImage": "gltfTextureSampler1.png"
-        },
-        {
-            "title": "GLTF Alien",
-            "playgroundId": "#XN37SR#5",
-            "referenceImage": "gltfAlien.png"
-        }
-    ]
+  "root": "https://cdn.babylonjs.com",
+  "tests": [
```
Comment (CedricGuillemet) [inline Apps/ValidationTests/Scripts/config.json:3]: Can you please add a PG test for this feature? I'm not sure this Canvas feature is tested on bjs.

**Review by CedricGuillemet — CHANGES_REQUESTED** [review]

File: azure-pipelines.yml (hunk 2 above)
Comment (SergioRZMasson) [reply azure-pipelines.yml]: CI changes were not intended for this PR, just reverted them.

File: Apps/ValidationTests/Scripts/config.json (hunk 1 above)
Comment (SergioRZMasson) [reply Apps/ValidationTests/Scripts/config.json]: Reverted changes.

File: Apps/ValidationTests/Scripts/config.json (hunk 4 above)
Comment (SergioRZMasson) [reply Apps/ValidationTests/Scripts/config.json]: @CedricGuillemet , just added a validation test to test clip() from canvas.

**Review by CedricGuillemet — APPROVED** [review]
LGTM!

**Pull request merged**